File: coyote/models/create_namespace_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel

from .retention import Retention
from .storage_type import StorageType


class CreateNamespaceIn(BaseModel):
    name: str

    retention: t.Optional[Retention] = None

    storage_type: t.Optional[StorageType] = Field(default=None, alias="storage_type")

File: src/models/create_namespace_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{retention::Retention, storage_type::StorageType};

#[non_exhaustive]
#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct CreateNamespaceIn {
    pub name: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub retention: Option<Retention>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub storage_type: Option<StorageType>,
}

impl CreateNamespaceIn {
    pub fn new(name: String) -> Self {
        Self {
            name,
            retention: None,
            storage_type: None,
        }
    }
}
